R/middleware.R

```diff
@@ -383,6 +385,30 @@ HandlerManager <- R6Class("HandlerManager",
   )
 )
 
+getResponseContentLength <- function(response, deleteOwnedContent) {
+  force(deleteOwnedContent)
+
+  result <- if (is.character(response$content)) {
+    nchar(response$content, type = "bytes")
+  } else if (is.raw(response$content)) {
+    length(response$content)
+  } else if (is.list(response$content) && !is.null(response$content$file)) {
+    if (deleteOwnedContent && isTRUE(response$content$owned)) {
+      on.exit(unlink(response$content$file, recursive = FALSE, force = FALSE), add = TRUE)
+    }
+    file.info(response$content$file)$size
+  } else {
+    warning("HEAD request for unexpected content class ", class(response$content)[[1]])
+    NULL
+  }
+
+  if (is.na(result)) {
```

Comment:
When would the `result` be `NA`? The cases I can think of are an `NA` string input, and when `file.info()` is called on a file that doesn't exist. A comment about this would be helpful so we remember.